src/daemon/PowerManager.cpp

```diff
@@ -174,8 +181,15 @@ const QString CK2_OBJECT = QStringLiteral("org.freedesktop.ConsoleKit.Manager");
         }
 
         void reboot() const {
-            if (!daemonApp->testing())
-                m_interface->call(QStringLiteral("Reboot"), true);
+            if (!daemonApp->testing()) {
+                if (m_interface->service() == LOGIN1_SERVICE) {
+                    qDebug() << "sddm calling RebootWithFlags";
+                    m_interface->call(QStringLiteral("RebootWithFlags"),
```

Comment:
It's weird that we're updating the interface, only to the use a string interface that doesn't use generated code.

```
m_interface->call(QStringLiteral("RebootWithFlags"),
```

If you've gone to the effort of updating the XML we may as well do:

```
auto message = m_interface->RebootWithFlags();
message << LOGIN1_FLAGS_KEXEC;
m_interface->connection()->call(message);
```